lib/src/benchmark_base.dart

```diff
@@ -62,3 +62,64 @@ class BenchmarkBase {
     emitter.emit(name, measure());
   }
 }
+
+class AsyncBenchmarkBase {
+  final String name;
+  final ScoreEmitter emitter;
+
+  // Empty constructor.
+  const AsyncBenchmarkBase(this.name, {this.emitter = const PrintEmitter()});
+
+  // The benchmark code.
+  // This function is not used, if both [warmup] and [exercise] are overwritten.
+  run() async {}
+
+  // Runs a short version of the benchmark. By default invokes [run] once.
+  void warmup() async {
```

Comment:
Future<void>, and likewise above and the other places

Reply:
Done